From d7fddfed6eff3f0274482f69827d9d93914da480 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Mon, 9 Feb 2015 15:43:00 +0000
Subject: Enable compiler warnings for libyelp

---
 libyelp/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libyelp/Makefile.am b/libyelp/Makefile.am
index 75606e0..5ebf22d 100644
--- a/libyelp/Makefile.am
+++ b/libyelp/Makefile.am
@@ -47,6 +47,7 @@ EXTRA_DIST += yelp-bz2-decompressor.c
 endif
 
 libyelp_la_CFLAGS =                             \
+	$(WARN_CFLAGS)				\
 	$(YELP_CFLAGS)                          \
 	-Wno-deprecated-declarations		\
 	-DDATADIR=\""$(datadir)"\"              \
-- 
cgit v0.12

