From 67c54ffc7a0e31af19b8cfcc784e203b71a0e07d Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Thu, 19 Feb 2015 14:23:54 +0100
Subject: [PATCH 111/315] local-display-factory: fix daemon/WaylandEnable

Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
---
 daemon/gdm-local-display-factory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/gdm-local-display-factory.c b/daemon/gdm-local-display-factory.c
index 95be1a82..ba01b2f5 100644
--- a/daemon/gdm-local-display-factory.c
+++ b/daemon/gdm-local-display-factory.c
@@ -477,7 +477,7 @@ static gboolean gdm_local_display_factory_sync_seats (GdmLocalDisplayFactory *fa
                         is_initial = TRUE;
 #ifdef ENABLE_WAYLAND_SUPPORT
                         gboolean wayland_enabled = FALSE;
-                        if (gdm_settings_direct_get_boolean (GDM_KEY_INITIAL_SETUP_ENABLE, &wayland_enabled)) {
+                        if (gdm_settings_direct_get_boolean (GDM_KEY_WAYLAND_ENABLE, &wayland_enabled)) {
                                 if (wayland_enabled) {
                                         session_type = "wayland";
                                 }
-- 
2.15.1

