From c2330e7fd167618e9ec2c7f8137e545e328805f1 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 29 Sep 2014 23:33:33 -0400
Subject: [PATCH] Add 3.16 version macros

---
 gdk/gdkversionmacros.h.in | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/gdk/gdkversionmacros.h.in b/gdk/gdkversionmacros.h.in
index 5601b12036..df4fe19fe2 100644
--- a/gdk/gdkversionmacros.h.in
+++ b/gdk/gdkversionmacros.h.in
@@ -132,6 +132,16 @@
  */
 #define GDK_VERSION_3_14        (G_ENCODE_VERSION (3, 14))
 
+/**
+ * GDK_VERSION_3_16:
+ *
+ * A macro that evaluates to the 3.16 version of GDK, in a format
+ * that can be used by the C pre-processor.
+ *
+ * Since: 3.16
+ */
+#define GDK_VERSION_3_16        (G_ENCODE_VERSION (3, 16))
+
 /* evaluates to the current stable version; for development cycles,
  * this means the next stable target
  */
@@ -316,5 +326,11 @@
 # define GDK_AVAILABLE_IN_3_14                _GDK_EXTERN
 #endif
 
+#if GDK_VERSION_MAX_ALLOWED < GDK_VERSION_3_16
+# define GDK_AVAILABLE_IN_3_16                GDK_UNAVAILABLE(3, 16)
+#else
+# define GDK_AVAILABLE_IN_3_16                _GDK_EXTERN
+#endif
+
 #endif  /* __GDK_VERSION_MACROS_H__ */
 
-- 
GitLab

